odoo/models.py

```diff
@@ -5926,7 +5926,7 @@ def _has_onchange(self, field, other_fields):
             presence of ``other_fields``.
         """
         return (field.name in self._onchange_methods) or any(
-            dep in other_fields for dep in self._dependent_fields(field)
+            dep in other_fields for dep in self._dependent_fields(field.base_field)
```

Comment:
Here are the dependencies we have in this case:
* `payment.tag_id` depends on `payment.move_id.tag_id` (inheritance)
* `payment.tag_name` depends on `payment.move_id.tag_name` (inheritance)
* `payment.move_id.tag_name` depends on `payment.move_id.tag_id` (computation)

Before this change, in order to trigger an onchange when `payment.tag_id` is modified, we determine whether some field in the view depends on `payment.tag_id`.  In our case, *no* field depends on `payment.tag_id`.  Instead, `payment.tag_name` indirectly depends on `payment.move_id.tag_id`, which by inheritance is equal to `payment.tag_id`.  This changes fixes the `on_change` flag in the view for this case.

Note that this does not discard any existing dependency: if some field depends on `payment.tag_id`, it also depends on `payment.move_id.tag_id`, because `_dependent_fields()` also returns indirect depend fields.

Reply:
Ok thanks for the explanation, your last point answers my question 👍🏽